lib/java/com/google/android/material/slider/BaseSlider.java

```diff
@@ -1476,22 +1488,62 @@ public void setTickInactiveTintList(@NonNull ColorStateList tickColor) {
   /**
    * Returns whether the tick marks are visible. Only used when the slider is in discrete mode.
    *
-   * @see #setTickVisible(boolean)
    * @attr ref com.google.android.material.R.styleable#Slider_tickVisible
+   * @deprecated Use {@link BaseSlider#getTickVisibilityMode()} to get the current
+   *     tick visibility mode or {@link BaseSlider#areTicksDrawn()} to get whether the ticks
+   *     are currently drawn.
    */
+  @Deprecated
   public boolean isTickVisible() {
-    return tickVisible;
+    switch (tickVisibilityMode) {
+      case TICK_VISIBILITY_AUTO_LIMIT:
+        return true;
+      case TICK_VISIBILITY_HIDDEN:
+        return false;
+      default:
+        return areTicksDrawn();
+    }
+  }
+
+  /**
+   * Returns whether the tick marks are drawn.
+   */
+  public boolean areTicksDrawn() {
```

Comment:
let's not expose it for now.

Reply:
Will `@RestrictTo(LIBRARY_GROUP)` be enough?

Reply:
I'll prefer not to expose the method at all, unless we have a strong reason to do so...

Reply:
In the `TICK_VISIBILITY_AUTO_HIDE` mode, ticks can be either visible or invisible, and we need to distinguish one state from another to create a set of uniform sliders.

Reply:
To be honest, I wouldn't like to hide or get rid of `areTicksDrawn()`, because it is more useful and returns the correct values at any time for any slider type (unlike `isTickVisible()` and `getTickVisibilityMode()` that exist mainly for symmetry with their setters).

However, I am also open to alternative solutions to the above problem.

Reply:
My concern is that "areTicksDrawn()" is deeply correlated to the underlying implementation rather than a more general concept. Also it's return value is kind of unpredictable (what if the slider has not been drawn yet?) I feel it's a bit risky to expose it because we can never remove it once we release it. 

Maybe using tickCount here instead of the drawn flag here will be better? I'm not sure.

Reply:
> Also it's return value is kind of unpredictable (what if the slider has not been drawn yet?)

If the slider hasn't been drawn yet, the ticks obviously haven't been drawn either, so `areTicksDrawn()` will return `false`.

Reply:
Well, ok, how about something like this:
```
/**
 * Returns whether the tick marks are visible. Only used when the slider is in discrete mode.
 */
public boolean isTickVisible() {
  switch (tickVisibilityMode) {
    case TICK_VISIBILITY_VISIBLE_ALL:
    case TICK_VISIBILITY_AUTO_LIMIT:
      return true;
    case TICK_VISIBILITY_AUTO_HIDE:
      if (stepSize == 0 || trackHeight == 0) {
        throw new IllegalStateException("Can't figure out the tick visibility yet.");
      }

      return getDesiredTickCount() <= getMaxTickCount();
    case TICK_VISIBILITY_HIDDEN:
      return false;
    default:
      throw new RuntimeException("isTickVisible() is not implemented for tickVisibilityMode=" + tickVisibilityMode);
  }
}
```

Reply:
Good, let's try.